### PR TITLE
gcc 14.1.0 built for multiple glibc versions

### PR DESCRIPTION
**Pull request by satmandu**
- GCC 14.1.0 has been built on Glibc versions 2.23 (i686 only), 2.27, and 2.37.
- Glibc 2.35 fallback is the 2.27 version.
- Updates `libnghttp2` => 1.59.0 (wanted to make sure compiler worked.)

This was created to avoid these errors:

```bash
Running Gnome post_installs for adwaita_icon_theme, at_spi2_core, cantarell_fonts, gdk_pixbuf, gtk3, gtk4, libcloudproviders, librsvg, libtiff, libwebp, and vte.
gtk-update-icon-cache: Cache file created successfully. 
g_module_open() failed for /usr/local/lib64/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so: /usr/local/lib64/libstdc++.so.6: undefined symbol: strtof128, version GLIBC_2.26
```

- Updated `install.sh` so the installer works with the gcc_lib packages in this PR.

### Run the following to get this pull request's changes locally for testing.
```bash
CREW_REPO=https://github.com/satmandu/chromebrew.git CREW_BRANCH=gcc crew update ; yes | crew upgrade
```

### CODE REVIEW

File: packages/libssp.rb [hunk 1]
```diff
@@ -4,22 +4,33 @@
...
+  version "14.1.0-glibc#{LIBC_VERSION}" # Do not use @_ver here, it will break the installer.
   license 'GPL-3, LGPL-3, libgcc, FDL-1.2'
   # When upgrading gcc_build, be sure to upgrade gcc_lib, gcc_dev, and libssp in tandem.
-  puts "#{self} version differs from gcc version #{Gcc_build.version}".orange if version.to_s.gsub(/-.*/,
-                                                                                                   '') != Gcc_build.version
+  puts "#{self} version (#{version}) differs from gcc version #{Gcc_build.version}".orange if version.to_s != Gcc_build.version
   compatibility 'all'
   source_url 'https://github.com/gcc-mirror/gcc.git'
-  git_hashtag "releases/gcc-#{version}"
+  git_hashtag 'releases/gcc-14.1.0'
```
Comment (uberhacker) [inline packages/libssp.rb:13]: `git_hashtag "releases/gcc-#{version.split('-').first}"`?

Comment (satmandu) [reply packages/libssp.rb]: Changed.

File: packages/gcc_build.rb [hunk 2]
```diff
@@ -3,19 +3,31 @@
 class Gcc_build < Package
   description 'The GNU Compiler Collection includes front ends for C, C++, Objective-C, Fortran, Ada, and Go.'
   homepage 'https://www.gnu.org/software/gcc/'
-  version '13.2.0' # Do not use @_ver here, it will break the installer.
+  version "14.1.0-glibc#{LIBC_VERSION}" # Do not use @_ver here, it will break the installer.
   license 'GPL-3, LGPL-3, libgcc, FDL-1.2'
   compatibility 'all'
   source_url 'https://github.com/gcc-mirror/gcc.git'
-  git_hashtag 'releases/gcc-13.2.0'
+  ggit_hashtag "releases/gcc-#{version.split('-').first}"
```
Comment (uberhacker) [inline packages/gcc_build.rb:10]: Extra `g`?

Comment (satmandu) [reply packages/gcc_build.rb]: Fixed

**Review by uberhacker — APPROVED** [review]

**Comment by Zopolis4** [comment]
Still getting similar errors:
```
FAILED: help/de/accounts.page help/de/archive.page help/de/bugs.page help/de/contributing.page help/de/index.page help/de/label.page help/de/limits.page help/de/overview.page help/de/preferences.page help/de/search.page help/de/shortcuts.page help/de/star.page help/de/write.page 
/usr/local/bin/itstool -m help/de/geary-de.gmo --lang de -o help/de ../help/C/accounts.page ../help/C/archive.page ../help/C/bugs.page ../help/C/contributing.page ../help/C/index.page ../help/C/label.page ../help/C/limits.page ../help/C/overview.page ../help/C/preferences.page ../help/C/search.page ../help/C/shortcuts.page ../help/C/star.page ../help/C/write.page
Traceback (most recent call last):
  File "/usr/local/bin/itstool", line 27, in <module>
    import libxml2
  File "/usr/local/lib64/python3.12/site-packages/libxml2.py", line 1, in <module>
    import libxml2mod
ImportError: /usr/local/lib64/libstdc++.so.6: undefined symbol: strtof128
```

**Pull request merged**